### DISCUSSION TITLE
How to use callService on version 0.29.0-dev？And where to get docs about this version?

### DISCUSSION THREAD

**Discussion by MiaoQ** [foxglove/community#783] [Q&A] [answered]
How to use context.callService method on version 0.29.0-dev shown on this docs？
https://foxglove.dev/docs/studio/extensions/panel-api#callservice

**Comment by achim-k** [comment, marked as answer]
You can use this method in a [custom panel](https://foxglove.dev/docs/studio/extensions/guide-create-custom-panel). An example panel for calling services which uses this method can be found here: https://github.com/foxglove/create-foxglove-extension/tree/main/examples/call-service-panel-example

**Reply by MiaoQ** [reply]
But I can't find callService method in PanelExtensionContext on version 0.29.0-dev of foxglove-studio. The source code likes below:


`export type PanelExtensionContext = {
    /**
     * The root element for the panel. Add your panel elements as children under this element.
     */
    readonly panelElement: HTMLDivElement;

    /**
     * Initial panel state
     */
    readonly initialState: unknown;

    /** Actions the panel may perform related to the user's current layout. */
    readonly layout: LayoutActions;

    /**
     * Subscribe to updates on this field within the render state. Render will only be invoked when
     * this field changes.
     */
    watch: (field: keyof RenderState) => void;

    /**
     * Save arbitrary object as persisted panel state. This state is persisted for the panel
     * within a layout.
     *
     * The state value should be JSON serializable.
     */
    saveState: (state: Partial<unknown>) => void;

    /**
     * Set the value of parameter name to value.
     *
     * @param name The name of the parameter to set.
     * @param value The new value of the parameter.
     */
    setParameter: (name: string, value: ParameterValue) => void;

    /**
     * Set the active preview time. Setting the preview time to undefined clears the preview time.
     */
    setPreviewTime: (time: number | undefined) => void;

    /**
     * Seek playback to the given time. Behaves as if the user had clicked the playback bar to seek.
     */
    seekPlayback?: (time: number) => void;

    /**
     * Subscribe to an array of topic names.
     *
     * Subscribe will update the current subscriptions to the list of topic names. Passing an empty
     * array will unsubscribe from all topics.
     */
    subscribe(topics: string[]): void;

    /**
     * Unsubscribe from all topics.
     */
    unsubscribeAll(): void;

    /**
     * Subscribe to any changes in application settings for an array of setting names.
     */
    subscribeAppSettings(settings: string[]): void;

    /**
     * Indicate intent to advertise on a specific topic and datatype.
     *
     * The options object is passed to the current data source for additional configuration.
     */
    advertise?(topic: string, datatype: string, options?: Record<string, unknown>): void;

    /**
     * Indicate that you no longer want to advertise on this topic.
     */
    unadvertise?(topic: string): void;

    /**
     * Publish a message on a given topic. You must first advertise on the topic before publishing.
     *
     * @param topic The name of the topic to publish the message on
     * @param message The message to publish
     */
    publish?(topic: string, message: unknown): void;

    /**
     * Process render events for the panel. Each render event receives a render state and a done callback.
     * Render events occur frequently (60hz, 30hz, etc).
     *
     * The done callback should be called once the panel has rendered the render state.
     */
    onRender?: (renderState: Readonly<RenderState>, done: () => void) => void;
  };`

**Reply by jtbandes** [reply]
Update to the latest version: https://foxglove.dev/download

**Reply by MiaoQ** [reply]
We have many customized development on version 0.29.0-dev, is there a recommended way to update version?

**Reply by jtbandes** [reply]
While you are welcome to do custom development on top of the open source project, we don’t generally provide support for any custom development except for our enterprise customers.

**Reply by MiaoQ** [reply]
Ok, thanks.
